src/System.Net.Requests/tests/HttpWebRequestTest.cs

```diff
@@ -1531,6 +1585,98 @@ public void MediaType_SetThenGet_ValuesMatch(Uri remoteServer)
             Assert.Equal(MediaType, request.MediaType);
         }
 
+        [Theory, MemberData(nameof(CachableWebRequestParameters))]
+        public void GetResponseAsync_CachableParameters_GotSuccessfully(HttpWebRequestParameters requestParameters)
+        {
+            RemoteExecutor.Invoke(async (serializedParameters) =>
+            {
+                var parameters = JsonSerializer.Deserialize<HttpWebRequestParameters>(serializedParameters);
+                var options = new LoopbackServer.Options { UseSsl = false };
+                ServicePointManager.SecurityProtocol = (SecurityProtocolType)parameters.SslProtocols;
+                ServicePointManager.CheckCertificateRevocationList = parameters.CheckCertificateRevocationList;
+
+                await LoopbackServer.CreateClientAndServerAsync(async uri =>
+                {
+                    HttpWebRequest request = WebRequest.CreateHttp(uri);
+                    parameters.Configure(request);
+                    request.Method = HttpMethod.Get.Method;
+
+                    using (WebResponse response = await request.GetResponseAsync())
+                    using (Stream myStream = response.GetResponseStream())
+                    using (var sr = new StreamReader(myStream))
+                    {
+                        string strContent = sr.ReadToEnd();
+                        Assert.True(strContent.Length > 0);
+                    }
+                }, server => server.AcceptConnectionAsync(async (con) =>
+                {
+                    await con.SendResponseAsync(content: RequestBody);
+
+                    StringBuilder sb = new StringBuilder();
+                    byte[] buf = new byte[1024];
+                    int count = 0;
+
+                    do
+                    {
+                        count = con.Stream.Read(buf, 0, buf.Length);
+                        if (count != 0)
+                        {
+                            sb.Append(Encoding.UTF8.GetString(buf, 0, count));
+                        }
+                    } while (count > 0 && sb.ToString().Substring(sb.Length - 4, 4) != "\r\n\r\n");
+
+                    Assert.StartsWith("GET / HTTP/1.1", sb.ToString());
+                }), options);
+                return RemoteExecutor.SuccessExitCode;
+            }, JsonSerializer.Serialize(requestParameters));
+        }
+
+        [Theory, MemberData(nameof(CachableWebRequestParameters))]
+        public void GetResponseAsync_NonCachableParameters_GotSuccessfully(HttpWebRequestParameters requestParameters)
```

Comment:
Both this test and the one above are using `CachableWebRequestParameters`... what's the difference between these tests?

Reply:
The difference is in the following line which disables caching because the delegate is not null.
`ServicePointManager.ServerCertificateValidationCallback = delegate { return true; };

Reply:
Ok.  Rather than entirely duplicating the test, it would be better to parameterize it with an extra Boolean indicating whether or not to set non-cacheable parameters.

Reply:
Also, how is the test validating the caching is or is not used?

Reply:
Currently, it does not validate it because we don't use any factory for HttpClient here and I cannot change HttpWebRequest contract to introduce one. So, I just checked cache usage under the debugger and via perf testing.
If you have any good idea about how to check HttpClient instantiation/retrieving from cache, could you please share it?

Reply:
We have existing tests like these:
https://github.com/dotnet/corefx/blob/dcc9f626eeb0751d4ba4499f623aac42d2e95c10/src/System.Net.Http/tests/FunctionalTests/SocketsHttpHandlerTest.cs#L1432
that validate that a connection gets reused or not based on relevant circumstances.